draft-ietf-quic-recovery.md

```diff
@@ -1608,17 +1615,20 @@ Invoked when DetectAndRemoveLostPackets deems packets lost.
 
 ~~~
    InPersistentCongestion(largest_lost):
-     // Persistent congestion cannot be declared on the
-     // first RTT sample.
-     if (is first RTT sample):
-       return false
      pto = smoothed_rtt + max(4 * rttvar, kGranularity) +
        max_ack_delay
-     congestion_period = pto * kPersistentCongestionThreshold
+     pc_period = pto * kPersistentCongestionThreshold
+
+     // Only look for persistent congestion if the period
+     // starts after getting the first RTT sample.
+     assert(first_rtt_sample != 0)
+     if (largest_lost.time_sent - pc_period > first_rtt_sample):
+       return false
+
      // Determine if all packets in the time period before the
      // largest newly lost packet, including the edges and
      // across all packet number spaces, are marked lost.
-     return AreAllPacketsLost(largest_lost, congestion_period)
+     return AreAllPacketsLost(largest_lost, pc_period)
```

Comment:
This needs to change. The idea now is that there's a window of packet loss that is pc_period long in time. The comment up there needs to change too.

How about:
```
// Determine if all packets were lost in a period long enough
// to declare persistent congestion.
return AreAllPacketsLostInPeriod(lost_packets, pc_period)
```

You'll have to change the parameter to `InPersistentCongestion()` to be `lost_packets`, but that's easily done, since this function is called from `OnPacketsLost()`.

Reply:
Because lost_packets is from a single PN space, I believe passing it in is insufficient and potentially misleading.